### PR TITLE
[Development] Add AdamLBFGS Optimizer

### LINKED ISSUE #12: Add Adam-LBFGS Optimizer
### Describe the feature

Adam-LBFGS is a hybrid optimization approach that combines Adam and L-BFGS, as shown in [this paper](https://arxiv.org/abs/2402.01868) and many others in the PINNs literature. The training starts with Adam and, at a specified epoch, it transitions to L-BFGS. The study finds that a combined Adam+L-BFGS approach outperforms using either optimizer individually, with the idea being that Adam is able to explore the full optimization landscape without getting stuck in local minima while LBFGS is used at the end to refine the optimization.

### It should be implemented because

Introducing Adam-LBFGS into `perceptrain` would provide a more robust optimization strategy for PINNs and other applications where Adam struggles with ill-conditioned loss landscapes.

### Additional context

A possible implementation could involve defining a new optimizer (defined on top of `torch.optim.Optimizer`) whose `step` method wraps around both Adam and L-BFGS `step` methods, switching between them based on the training epoch. 

An additional useful feature could be, instead of requiring the user to manually specify the switching epoch, an automated approach where the transition occurs when Adam stops making significant progress.

### Would you like to work on this issue?

Yes

### PR DESCRIPTION
**Pull request by DanieleCucurachi**
Closes #12 .

### CODE REVIEW

**Review by chMoussa — APPROVED** [review]
LGTM
